python_modules/dagster/dagster/_core/storage/fs_io_manager.py

```diff
@@ -18,8 +23,110 @@
+    * A temporary directory.
+
+    Assigns each op output to a unique filepath containing run ID, step key, and output name.
+    Assigns each asset to a single filesystem path, at "<base_dir>/<asset_key>". If the asset key
+    has multiple components, the final component is used as the name of the file, and the preceding
+    components as parent directories under the base_dir.
+
+    Subsequent materializations of an asset will overwrite previous materializations of that asset.
+    So, with a base directory of "/my/base/path", an asset with key
+    `AssetKey(["one", "two", "three"])` would be stored in a file called "three" in a directory
+    with path "/my/base/path/one/two/".
+
+    Example usage:
+
+
+    1. Attach an IO manager to a set of assets using the reserved resource key ``"io_manager"``.
+
+    .. code-block:: python
+
+        from dagster import Definitions, asset, FilesystemIOManager
+
+        @asset
+        def asset1():
+            # create df ...
+            return df
+
+        @asset
+        def asset2(asset1):
+            return asset1[:5]
+
+        defs = Definitions(
+            assets=[asset1, asset2],
+            resources={
+                "io_manager": FilesystemIOManager(base_dir="/my/base/path")
+            },
+        )
+
+
+    2. Specify a job-level IO manager using the reserved resource key ``"io_manager"``,
+    which will set the given IO manager on all ops in a job.
+
+    .. code-block:: python
+
+        from dagster import FilesystemIOManager, job, op
+
+        @op
+        def op_a():
+            # create df ...
+            return df
+
+        @op
+        def op_b(df):
+            return df[:5]
+
+        @job(
+            resource_defs={
+                "io_manager": FilesystemIOManager(base_dir="/my/base/path")
+            }
+        )
+        def job():
+            op_b(op_a())
+
+
+    3. Specify IO manager on :py:class:`Out`, which allows you to set different IO managers on
+    different step outputs.
+
+    .. code-block:: python
+
+        from dagster import FilesystemIOManager, job, op, Out
+
+        @op(out=Out(io_manager_key="my_io_manager"))
+        def op_a():
+            # create df ...
+            return df
+
+        @op
+        def op_b(df):
+            return df[:5]
+
+        @job(resource_defs={"my_io_manager": FilesystemIOManager()})
+        def job():
+            op_b(op_a())
+
+    """
+
+    base_dir: Optional[str] = Field(default=None, description="Base directory for storing files.")
+
+    def create_io_manager(self, context: InitResourceContext) -> "PickledObjectFilesystemIOManager":
+        base_dir = self.base_dir or check.not_none(context.instance).storage_directory()
+        return PickledObjectFilesystemIOManager(base_dir=base_dir)
+
+
 @io_manager(
-    config_schema={"base_dir": Field(StringSource, is_required=False)},
+    config_schema={"base_dir": DagsterField(StringSource, is_required=False)},
```

Comment:
`FilesystemIOManager.to_config_schema()` and you can avoid `DagsterField` methinks

Reply:
yup, will update - initially structured the diff to be conservative and not touch any fs_io_manager codepaths in case we wanted to try to slip this into 1.3, but I think we'll just wait for the dot release